packages/core/src/command/consolidated/consolidated-get.ts

```diff
@@ -0,0 +1,47 @@
+import { Bundle, Resource } from "@medplum/fhirtypes";
+import { createConsolidatedDataFilePath } from "../../domain/consolidated/filename";
+import { executeWithRetriesS3, returnUndefinedOn404, S3Utils } from "../../external/aws/s3";
+import { Config } from "../../util/config";
+
+const defaultS3RetriesConfig = {
+  maxAttempts: 3,
+  initialDelay: 500,
+};
+
+export type Consolidated = {
+  bundle: Bundle<Resource> | undefined;
+  fileLocation: string;
+  fileName: string;
+};
+
+export async function getConsolidated({
+  cxId,
+  patientId,
+  fileLocation = Config.getMedicalDocumentsBucketName(),
+  region = Config.getAWSRegion(),
+  log = console.log,
+}: {
+  cxId: string;
+  patientId: string;
+  fileLocation?: string;
+  region?: string;
+  log?: typeof console.log;
+}): Promise<Consolidated> {
+  const s3Utils = new S3Utils(region);
+  const fileName = createConsolidatedDataFilePath(cxId, patientId);
+  const consolidatedDataRaw = await executeWithRetriesS3<string | undefined>(
+    async () => returnUndefinedOn404(() => s3Utils.getFileContentsAsString(fileLocation, fileName)),
+    { ...defaultS3RetriesConfig, log }
+  );
+  const bundle = parseConsolidatedRaw(consolidatedDataRaw, log);
+  return { bundle, fileLocation, fileName };
+}
+
+function parseConsolidatedRaw(
+  contents: string | undefined,
+  log: typeof console.log
+): Bundle | undefined {
+  if (!contents) return undefined;
+  log(`Converting payload to JSON, length ${contents.length}`);
+  return JSON.parse(contents) as Bundle;
```

Comment:
can we use a type guard isntead of casting? 

```function parseConsolidatedRaw(
  contents: string | undefined,
  log: typeof console.log
): Bundle | undefined {
  if (!contents) return undefined;
  log(`Converting payload to JSON, length ${contents.length}`);
  try {
    const parsed = JSON.parse(contents);
    if (isBundle(parsed)) {
      return parsed;
    } else {
      log(`Parsed object is not a valid Bundle.`);
      return undefined;
    }
  } catch (error) {
    log(`Failed to parse JSON: ${(error as Error).message}`);
    return undefined;
  }
}

export function isBundle(obj: any): obj is Bundle {
  return obj && typeof obj === "object" && obj.resourceType === "Bundle";
}

Reply:
yeah, I'll add a function for this, there's a bunch of places we do that thing.

Reply:
kk is this for this PR or the follow on one that you will make this change?

Reply:
The next one, I already have part of it coded, just don't want to add much changes to this one to avoid slowing down the release any further